backend/dataall/modules/shares_base/services/sharing_service.py

```diff
@@ -356,6 +362,48 @@ def reapply_share(cls, engine: Engine, share_uri: str) -> bool:
                     if not lock_released:
                         log.error(f'Failed to release lock for dataset {share_data.dataset.datasetUri}.')
 
+    @staticmethod
```

Comment:
Already commented above, `fetch_pending_shares` should be part of a db.repository

Reply:
Moved it there.